common/lanelet2_extension/lib/SignalizedIntersection.cpp

```diff
@@ -0,0 +1,191 @@
+
+    ConstLanelets SignalizedIntersection::getExitLanelets() const {return getParameters<ConstLanelet>(CarmaRoleNameString::IntersectionExit);}
+
+    ConstLanelets SignalizedIntersection::getInteriorLanelets() const {return getParameters<ConstLanelet>(CarmaRoleNameString::IntersectionInterior);}
+
+    std::unique_ptr<lanelet::RegulatoryElementData> SignalizedIntersection::buildData(Id id, const Lanelets& entry, const Lanelets& exit,
+                                                                const Lanelets& interior)
+    {
+        // Add parameters
+        RuleParameterMap rules;
+        rules[lanelet::CarmaRoleNameString::IntersectionEntry].insert(rules[lanelet::CarmaRoleNameString::IntersectionEntry].end(), entry.begin(),
+                                                        entry.end());
+        rules[lanelet::CarmaRoleNameString::IntersectionExit].insert(rules[lanelet::CarmaRoleNameString::IntersectionExit].end(), exit.begin(),
+                                                        exit.end());
+        rules[lanelet::CarmaRoleNameString::IntersectionInterior].insert(rules[lanelet::CarmaRoleNameString::IntersectionInterior].end(), interior.begin(),
+                                                        interior.end());
+
+        // Add attributes
+        AttributeMap attribute_map({ { AttributeNamesString::Type, AttributeValueString::RegulatoryElement },
+                                    { AttributeNamesString::Subtype, RuleName }});
+
+
+        return std::make_unique<RegulatoryElementData>(id, rules, attribute_map);
+    }
+
+    std::vector<CarmaTrafficSignalConstPtr> SignalizedIntersection::getTrafficSignals(const ConstLanelet& llt) const
+    {
+        return llt.regulatoryElementsAs<CarmaTrafficSignal>();
+    }
+
+    void SignalizedIntersection::addLanelet(Lanelet lanelet, IntersectionSection section)
+    {
+        switch (section)
+        {
+            case IntersectionSection::ENTRY:
+                parameters()[CarmaRoleNameString::IntersectionEntry].emplace_back(RuleParameter(lanelet));
+                break;
+            case IntersectionSection::EXIT:
+                parameters()[CarmaRoleNameString::IntersectionExit].emplace_back(RuleParameter(lanelet));
+                break;
+            case IntersectionSection::INTERIOR:
+                parameters()[CarmaRoleNameString::IntersectionInterior].emplace_back(RuleParameter(lanelet));
+                break;
+            default:
+                throw std::invalid_argument("Invalid section is passed as IntersectionSection");
+        }
+        
+        section_lookup[lanelet.id()] = section;
+    }
+    
+    namespace {
+
+        template <typename T>
+        bool findAndErase(const T& primitive, RuleParameters* member)
+        {
+            if (member == nullptr)
+            {
+                std::cerr << __FUNCTION__ << ": member is null pointer";
```

Comment:
I think you should be able to 
```
#include <lanelet2_extension/logging/logger.h>
```
Then use 
```
LOG_ERROR_STREAM()
```
This is a macro I created for this package which is injected with the appropriate logger at build time, so it will use ROS_LOG_ERROR_STREAM

Reply:
oh cool. thanks. About the documentation, I was going to update after PR was done, but I will do it now